lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -240,8 +240,28 @@ fn mpp_to_one_hop_blinded_path() {
 		Some(payment_secret), ev.clone(), false, None);
 
 	let ev = remove_first_msg_event_to_node(&nodes[2].node.get_our_node_id(), &mut events);
-	pass_along_path(&nodes[0], expected_route[1], amt_msat, payment_hash.clone(),
+	let event = pass_along_path(&nodes[0], expected_route[1], amt_msat, payment_hash.clone(),
 		Some(payment_secret), ev.clone(), true, None);
+
+	match event.unwrap() {
+		Event::PaymentClaimable { mut via_channel_id_pairs, .. } => {
+			let channels = nodes[3].node.list_channels();
+
+			// Convert to HashSets for order-agnostic comparison
+			let expected_chan_ids: HashSet<_> = channels.iter().map(|d| d.channel_id).collect();
+			let expected_user_chan_ids: HashSet<_> = channels.iter().map(|d| d.user_channel_id).collect();
+
+			let actual_chan_ids: HashSet<_> = via_channel_id_pairs.iter().map(|(chan_id, _)| *chan_id).collect();
+			let actual_user_chan_ids: HashSet<_> = via_channel_id_pairs.iter()
+				.filter_map(|(_, user_id_opt)| *user_id_opt)
+				.collect();
+
+			assert_eq!(actual_chan_ids, expected_chan_ids);
+			assert_eq!(actual_user_chan_ids, expected_user_chan_ids);
```

Comment:
Isn't it possible to do a vec<tuple> comparison here to avoid splitting everything into two lists?

Reply:
Thanks for the pointer, Joost!

Since `list_channels` returns channels in arbitrary order, I wanted the comparison to be order-agnostic.  
In [pr3655.12](https://github.com/shaavan/rust-lightning/commits/pr3655.12), I updated the code to compare sorted vectors of the expected and actual `inbound_channel_ids` to achieve that. It makes the test cleaner and easier to reason about.